app/src/main/java/fr/free/nrw/commons/contributions/ContributionsFragment.java

```diff
@@ -601,10 +611,22 @@ public void notifyDataSetChanged() {
     @Override
     public void retryUpload(Contribution contribution) {
         if (NetworkUtils.isInternetConnectionEstablished(getContext())) {
-            if (contribution.getState() == STATE_FAILED || contribution.getState() == STATE_PAUSED || contribution.getState()==Contribution.STATE_QUEUED_LIMITED_CONNECTION_MODE) {
-                contribution.setState(Contribution.STATE_QUEUED);
-                contributionsPresenter.saveContribution(contribution);
-                Timber.d("Restarting for %s", contribution.toString());
+            if (contribution.getState() == STATE_PAUSED || contribution.getState()==Contribution.STATE_QUEUED_LIMITED_CONNECTION_MODE) {
+                restartUpload(contribution);
+            } else if (contribution.getState() == STATE_FAILED) {
```

Comment:
I was trying to test the change in my device. Blindly retrying all failed uploads does seem to be affecting the UX badly. The app seems to just retrying genuinely failed uploads again-and-again each time I open the app.

Is there really no other attribute that we have that could help with identifying if an upload failed for a genuine reason? (I'm presuming its because of beta cluster not supporting stashed uploads).

Also, I remember cancelling all uploads manually one time. The cancel didn't seem to have worked. Has anyone else observed this too?